help/wscript

```diff
@@ -9,50 +9,56 @@ import os
 from waflib import Logs, TaskGen
 
 def _read_makefile_am(filename):
-	"read a Makefile.am file for DOC_* variable definitions, return a dict"
+	"read a Makefile.am file for HELP_* variable definitions, return a dict"
 	varstring = open(filename).read()
 	varstring = varstring.replace("\\\n", " ")
-	varlines = [L for L in varstring.splitlines() if L.startswith("DOC")]
+	varlines = [L for L in varstring.splitlines() if L.startswith("HELP")]
 	return dict(tuple(map(str.strip, var.split("=", 1))) for var in varlines)
 
 def init_mallard(self):
 	mf_am = self.path.find_resource(self.variable_definitions)
-	DOC_VAR = _read_makefile_am(mf_am.abspath())
+	HELP_VAR = _read_makefile_am(mf_am.abspath())
 
-	require_vars = "DOC_ID DOC_LINGUAS DOC_PAGES".split()
-	have_vars = set(var for var in DOC_VAR if DOC_VAR[var])
+	require_vars = "HELP_ID HELP_LINGUAS HELP_FILES".split()
+	have_vars = set(var for var in HELP_VAR if HELP_VAR[var])
 	missing_vars = set(require_vars).difference(have_vars)
 	if missing_vars:
-		print("Missing DOC variable declarations in {}:".format(mf_am.abspath()))
+		print("Missing HELP variable declarations in {}:".format(mf_am.abspath()))
 		print("\n".join(missing_vars))
 
-	self.bld.env.update(DOC_VAR)
+	self.bld.env.update(HELP_VAR)
 	self.default_install_path='${PREFIX}/share/help'
 
 def apply_mallard(self):
 	bld = self.bld
-	lst = self.to_list(bld.env["DOC_LINGUAS"])
+	lst = self.to_list(bld.env["HELP_LINGUAS"])
 	cnode = self.path.find_dir("C")
 
 	pages = [p.name for p in cnode.ant_glob("*.page")]
 
 	# Check if the declared page list is consistent
-	declared_pages = self.to_list(bld.env["DOC_PAGES"])
+	declared_pages = self.to_list(bld.env["HELP_FILES"])
 	missing_pages = set(pages).difference(declared_pages)
 	if missing_pages:
 		print("Warning: Some pages not declared:")
 		print("\n".join(missing_pages))
 
-	install_path = lambda lang: os.path.join(bld.env.DATADIR, "help", lang, "${DOC_ID}")
+	install_path = lambda lang: os.path.join(bld.env.DATADIR, "help", lang, "${HELP_ID}")
 
 	for lang in lst:
 		node = self.path.find_resource("%s/%s.po" % (lang, lang))
+		mo = self.path.find_or_declare("%s/%s.mo" % (lang, lang))
+		bld(name='msgfmt', color='BLUE',
+		    rule='${MSGFMT} ${SRC} -o ${TGT}',
+		    source=node,
+		    target=mo,
+		    )
```

Comment:
Why is this msgfmt call added? It seems weird to me that this was not previously needed (as xml2po probably produces .po files, so I would expect a msg2fmt (which I think converts .po to .mo) call with xml2po, not with `itstool` (which I suspect converts to .mo directly?)

Reply:
It seems to me that itstool expects .mo files, and can't handle .po files directly. [The reference](http://itstool.org/documentation/basic-usage.html#merge) says that the po files need to be converted to mo files first.

Reply:
Ah, I see how it works now. Indeed, xml2po was previously used not to generate po files, but it *read* the po file and produced a translated XML. The bld commandline here is a bit confusing, because the -p option (for read .po file and translate xml) was hidden in XML2POFLAGS, and source contains *two* filenames, which are both passed in the place of `${SRC}`. So the old code would take XML and .po and produce translated xml. The new itstool takes XML and .mo and produces translated XML. So, this is clarified.